### ISSUE TITLE
Look-Ahead Bias, Cross-Validation, and Workflow

### ISSUE THREAD

**Issue by iewaij** [sonic171717/undernourishment_prediction#1]
Hey,

I have briefly skimmed the code and I think you are on a good track, here is what I would like to suggest:

1. Set a period aside as test period to avoid look-ahead bias. If I understand the task correctly, the dataset is a time series data. A major issue with time series data is that there is hidden information correlated with time which is not predictable. The model may learn to predict the time information rather than the true model if you simply shuffle your dataset with `train_test_split`. To make sure your model's test score represents the forecasting performance, set 2000-2018 as train set and 2018-2022 as test set for example. I would expect the test performance drop by a large margin, which is normal.
2. Use [cross-validations](https://scikit-learn.org/stable/modules/cross_validation.html) to evaluate model performances at development/training stage. You might need time series split.
3. Improve and document your workflow. Usually, a data analysis project starts with the data extraction, exploratory data analysis, data transformation and then modelling. Maybe have more notebooks reflecting your work on these parts. You can refer to the course's recommended books and [a project I did](https://iewaij.github.io/data-analytics-bank-marketing/bank_marketing.pdf).
4. Make sure people can reproduce your notebooks easily, e.g. be sure your file paths are readable. `pd.read_csv("data.csv")` is only executable when `data.csv` is in the same folder of your notebook. If you have a file structure like this:

        project/
            data/
                data.csv
            notebooks/
                eda.ipynb

In `eda.ipynb`, you can use `pd.read_csv("../data/data.csv")` to access the parent folder and the data file.

**Comment by nilsmart96** [comment]
We have been reviewing your points and implemented them. Thank you very much for your input!
